recon/src/protocol.rs

```diff
@@ -353,13 +358,18 @@ where
         // Use a stack size large enough to handle the split factor of range requests.
         let ranges_stack = Vec::with_capacity(PENDING_RANGES_LIMIT * 10);
 
+        // Generate a random sync ID that fits in a JSON object and can be used to correlate Recon sync conversations
+        // across Recon nodes.
+        let sync_id = rand::thread_rng().gen_range(1..9007199254740991);
```

Comment:
You could use the u64::MAX constant here maybe. Also you can put underscores in numerics (eg 1_000_000) to make it read like commas if it's helpful. I don't think it matters here though.